Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -258,6 +259,39 @@ void StringLiteral::generate_bytecode(Bytecode::Generator& generator) const
     generator.emit<Bytecode::Op::NewString>(generator.intern_string(m_value));
 }
 
+void RegExpLiteral::generate_bytecode(Bytecode::Generator& generator) const
+{
+    auto source_index = generator.intern_string(m_pattern);
+    u16 flags = 0;
+
+    for (auto ch : m_flags) {
+        switch (ch) {
+        case 'g':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Global);
+            break;
+        case 'i':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Insensitive);
+            break;
+        case 'm':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Multiline);
+            break;
+        case 's':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::SingleLine);
+            break;
+        case 'u':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Unicode);
+            break;
+        case 'y':
+            flags = flags | static_cast<u16>(ECMAScriptFlags::Sticky);
```

Comment:
PSA: You can now use `to_underlying` to case enum classes to their underlying type instead.